x-pack/test_serverless/functional/test_suites/common/platform_security/user_profiles/user_profiles.ts

```diff
@@ -8,37 +8,48 @@
 import expect from '@kbn/expect';
 import { FtrProviderContext } from '../../../../ftr_provider_context';
 
-export default function ({ getPageObjects }: FtrProviderContext) {
-  const pageObjects = getPageObjects(['svlCommonPage', 'common', 'userProfiles']);
+const VIEWER_ROLE = 'viewer';
+
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const pageObjects = getPageObjects(['svlCommonPage', 'common', 'svlUserProfile']);
+  const svlUserManager = getService('svlUserManager');
+  const find = getService('find');
 
   describe('User Profile Page', async () => {
     before(async () => {
-      // TODO: migrate to SAML role when profile page displays the data
-      await pageObjects.svlCommonPage.login();
+      await pageObjects.svlCommonPage.loginWithRole(VIEWER_ROLE);
     });
 
     after(async () => {
       await pageObjects.svlCommonPage.forceLogout();
     });
 
-    describe('Theme', async () => {
-      it('should change theme based on the User Profile Theme control', async () => {
+    describe('User details', async () => {
+      it('should display correct user deatils', async () => {
         await pageObjects.common.navigateToApp('security_account');
 
-        const themeKeyPadMenu = await pageObjects.userProfiles.getThemeKeypadMenu();
-        expect(themeKeyPadMenu).not.to.be(null);
+        const userData = await svlUserManager.getUserData(VIEWER_ROLE);
+
+        const actualFullname = await pageObjects.svlUserProfile.getProfileFullname();
+        const actualEmail = await pageObjects.svlUserProfile.getProfileEmail();
+
+        expect(actualFullname).to.be(userData.fullname);
+        expect(actualEmail).to.be(userData.email);
+      });
 
-        await pageObjects.userProfiles.changeUserProfileTheme('Dark');
-        const darkModeTag = await pageObjects.userProfiles.getThemeTag();
-        expect(darkModeTag).to.be('v8dark');
+      it('should not have edit actions', async () => {
+        const fullNameInputField = await find.allByCssSelector(
+          '[data-test-subj="userProfileFullName"]'
+        );
```

Comment:
__nit:__ we have a dedicated a service to query and interact with the elements with `data-test-subj` attribute: `const testSubjects = getService('testSubjects')`. Check it out! https://github.com/elastic/kibana/blob/b40b6d81f247945becfc91d8591da0dda6bc2e4b/packages/kbn-ftr-common-functional-ui-services/services/test_subjects.ts#L28

Reply:
As I've tested it, if I use `testSubjects ` it always awaits for element to be present and if it is not, the test itself fails.
That's why I'm checking it with `allByCssSelector` here

Reply:
Hmm, you mean even `testSubjects.findAll` will fail element isn't present?

Reply:
let me check, I was using find, not findAll

Reply:
pushed changes, thanks 👍